tests/phpunit/tests/modules/modules.php

```diff
@@ -0,0 +1,410 @@
+		$enqueued_before_registered->setAccessible( true );
+		$this->old_enqueued_before_registered = $enqueued_before_registered->getValue();
+		$enqueued_before_registered->setValue( null, array() );
+	}
+
+	public function tear_down() {
+		$registered = new ReflectionProperty( 'WP_Modules', 'registered' );
+		$registered->setAccessible( true );
+		$registered->setValue( null, $this->old_registered );
+
+		$enqueued_before_registered = new ReflectionProperty( 'WP_Modules', 'enqueued_before_registered' );
+		$enqueued_before_registered->setAccessible( true );
+		$enqueued_before_registered->setValue( null, $this->old_enqueued_before_registered );
+
+		parent::tear_down();
+	}
+
+	/**
+	 * Gets a list of the enqueued modules.
+	 *
+	 * @return array Enqueued module URLs, keyed by module identifier.
+	 */
+	public function get_enqueued_modules() {
+		$modules_markup   = get_echo( array( 'WP_Modules', 'print_enqueued_modules' ) );
+		$p                = new WP_HTML_Tag_Processor( $modules_markup );
+		$enqueued_modules = array();
+
+		while ( $p->next_tag(
+			array(
+				'tag'  => 'SCRIPT',
+				'type' => 'module',
+			)
+		) ) {
+			$enqueued_modules[ $p->get_attribute( 'id' ) ] = $p->get_attribute( 'src' );
+		}
+
+		return $enqueued_modules;
+	}
+
+	/**
+	 * Gets the modules listed in the import map.
+	 *
+	 * @return array Import map entry URLs, keyed by module identifier.
+	 */
+	public function get_import_map() {
+		$import_map_markup = get_echo( array( 'WP_Modules', 'print_import_map' ) );
+		preg_match( '/<script type="importmap">.*?(\{.*\}).*?<\/script>/s', $import_map_markup, $import_map_string );
+		return json_decode( $import_map_string[1], true )['imports'];
+	}
+
+	/**
+	 * Gets a list of preloaded modules.
+	 *
+	 * @return array Preloaded module URLs, keyed by module identifier.
+	 */
+	public function get_preloaded_modules() {
+		$preloaded_markup  = get_echo( array( 'WP_Modules', 'print_module_preloads' ) );
+		$p                 = new WP_HTML_Tag_Processor( $preloaded_markup );
+		$preloaded_modules = array();
+
+		while ( $p->next_tag(
+			array(
+				'tag' => 'LINK',
+				'rel' => 'modulepreload',
+			)
+		) ) {
+			$preloaded_modules[ $p->get_attribute( 'id' ) ] = $p->get_attribute( 'href' );
+		}
+
+		return $preloaded_modules;
+	}
+
+	/**
+	 * Tests that a module gets enqueued correctly after being registered.
+	 *
+	 * @ticket 56313
+	 *
+	 * @covers ::wp_register_module
+	 * @covers ::wp_enqueue_module
+	 * @covers WP_Modules::print_enqueued_modules
+	 */
+	public function test_wp_enqueue_module() {
+		wp_register_module( 'foo', '/foo.js' );
+		wp_register_module( 'bar', '/bar.js' );
+		wp_enqueue_module( 'foo' );
+		wp_enqueue_module( 'bar' );
+
+		$enqueued_modules = $this->get_enqueued_modules();
+
+		$this->assertEquals( 2, count( $enqueued_modules ) );
+		$this->assertEquals( true, str_starts_with( $enqueued_modules['foo'], '/foo.js' ) );
```

Comment:
Nit, assertions like that can be simplified with different helper methods:
```suggestion
		$this->assertTrue( str_starts_with( $enqueued_modules['foo'], '/foo.js' ) );
```

or:

```suggestion
		$this-> assertStringStartsWith( '/foo.js', $enqueued_modules['foo'] );
```

More in https://docs.phpunit.de/en/9.6/assertions.html.

Reply:
Done in https://github.com/WordPress/wordpress-develop/pull/5818/commits/e644da8072e242f3cb118d43f71410252f90d4af.